sdk/src/main/java/io/radar/sdk/RadarLocationManager.kt

```diff
@@ -573,7 +572,7 @@ internal class RadarLocationManager(
                 ) {
                     locationManager.replaceSyncedGeofences(nearbyGeofences)
 
-                    if (foregroundService != null && foregroundService.updatesOnly) {
+                    if (options.foregroundServiceEnabled && foregroundService.updatesOnly) {
```

Comment:
Looks like `updateTracking` does this by checking `RadarForegroundService.started`, we may want to do the same thing here?

Reply:
That check is a little bit different. There we're looking at `RadarForegroundService.started` to see if the service is currently running. Here we're looking at whether the foreground service should be started to begin with based on tracking options and foreground service options.

Reply:
I guess both places are calling `locationManager.stopForegroundService()`, trying understand why one should look at the `started` flag but the other one should look at configuration

Reply:
```suggestion
                    if (RadarForegroundService.started) {
```

Reply:
So `updatesOnly == true` will only show the foreground service while a location update is being processed. The logic for that lives in this function.

On the other hand, `updatesOnly == false` (default) will show the foreground service as long as tracking is on. The logic for that lives in the `updateTracking()` function.

If `updatesOnly == true`, we need to stop the foreground service when we finish processing a location update. That's the logic on line 575. With your suggested change, we would stop the foreground service after processing every location update, even if `updatesOnly == false`. That means the foreground service would go away in continuous mode after the first location update.

Reply:
ok got it